packages/@uppy/tus/src/index.js

```diff
@@ -252,6 +259,47 @@ module.exports = class Tus extends BasePlugin {
         resolve(upload)
       }
 
+      uploadOptions.onShouldRetry = (err, retryAttempt, options) => {
+        const status = err?.originalResponse?.getStatus()
+        if (status === 429) {
+          // HTTP 429 Too Many Requests => to avoid the whole download to fail, pause all requests.
+          if (!this.requests.isPaused) {
+            const next = this.#retryDelayIterator?.next()
+            if (next == null || next.done) {
+              return false
+            }
+            this.requests.limit = Math.ceil(this.requests.limit / 2)
```

Comment:
Not sure if this is the most efficient way of doing it, but it seems to mostly do the trick for now.